### PR TITLE
Improve performance when interpreting by buffering

### PR DESCRIPTION
**Pull request by jeffwilliams**
On large files interpret was really slow. This change speeds it up significantly by buffering.

### CODE REVIEW

File: interpret.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ var MaxEscapeCodeLen uint = 15
 func interpret(r io.ByteReader, w io.Writer, subst func(s string) []byte) error {
 	inEscape := false
 	escape := &bytes.Buffer{}
+	out := bufio.NewWriter(w)
+	defer out.Flush()
```
Comment (aybabtme) [inline interpret.go]: `out.Flush()` could error here, do you mind doing:
```go
func interpret(...) (err error) { // make `err` be named
	// ...
	defer func() { 
		if err == nil { 
			err = out.Flush() 
		} 
	}()
```

**Comment by aybabtme** [comment]
@jeffwilliams thanks a lot for this PR, it LGTM minus this very little thing about `Flush`

**Comment by jeffwilliams** [comment]
Sure thing, that's a good point. I'll make the change.

**Comment by aybabtme** [comment]
@jeffwilliams awesome, thanks again!

**Pull request merged**